### PR TITLE
planner/cascades: estimate the cost of `PhysicalProjection`

### LINKED ISSUE #14967: planner/cascades: estimate the cost of `PhysicalProjection` and `PhysicalWindow`
## Feature Request

**Is your feature request related to a problem? Please describe:**

1. We forgot to implement the `CalcCost` method for `ProjectionImpl` in cascades planner.

2. We didn't calculate the cost of `PhsyicalWindow` in both plannercore and cascades planenr.

**Describe the feature you'd like:**

1. Implement `CalcCost` method for `ProjectionImpl`.

2. Implement `CalcCost` method for `WindowImpl`, and implement `attach2Task` method for `PhysicalWindow`.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

estimate the cost of `PhysicalProjection`

Issue Number: ref #14967 

### What is changed and how it works?

Implement `CalcCost` method for `ProjectionImpl` and `attach2Task` method for PhysicalWindow.

### Release note 

- N/A

### CODE REVIEW

**Comment by Yisaer** [comment]
I will fix the unit test when the change get approved. PTAL when you are free. @francis0407

File: planner/implementation/simple_plans.go [hunk 1]
```diff
@@ -28,6 +28,12 @@ func NewProjectionImpl(proj *plannercore.PhysicalProjection) *ProjectionImpl {
 	return &ProjectionImpl{baseImpl{plan: proj}}
 }
 
+// CalcCost implements Implementation CalcCost interface.
+func (projection *ProjectionImpl) CalcCost(outCount float64, children ...memo.Implementation) float64 {
```
Comment (francis0407) [inline planner/implementation/simple_plans.go:32]:
```suggestion
func (impl *ProjectionImpl) CalcCost(outCount float64, children ...memo.Implementation) float64 {
```

File: planner/implementation/simple_plans.go [hunk 2]
```diff
@@ -28,6 +28,12 @@ func NewProjectionImpl(proj *plannercore.PhysicalProjection) *ProjectionImpl {
...
+	proj := projection.plan.(*plannercore.PhysicalProjection)
+	return proj.GetCost(children[0].GetPlan().Stats().RowCount) + children[0].GetCost()
```
Comment (francis0407) [inline planner/implementation/simple_plans.go:34]: We need to set `impl.cost = proj.GetCost(children[0].GetPlan().Stats().RowCount) + children[0].GetCost()`
and return `impl.cost`

Comment (Yisaer) [reply planner/implementation/simple_plans.go]: updated

**Comment by zz-jason** [comment]
No need to write release notes for cascades planner related PRs since cascades planner is not generally available to users.

**Pull request merged**